utilities/transactions/write_prepared_txn.cc

```diff
@@ -136,7 +144,8 @@ Status WritePreparedTxn::CommitInternal() {
     // Commit the data that is accompnaied with the commit marker
     // TODO(myabandeh): skip AddPrepared
     wpt_db_->AddPrepared(commit_seq);
```

Comment:
Is it possible here that the write become visible between `WriteImpl` and `AddPrepared` (because it is in memtable but not in prepare heap) and then non-visible between `AddPrepared` and `AddCommited` (because it appears to be non-committed prepared at the point)?

Reply:
Nice catch! Although the probability is almost zero, it is possible that max_evicted_seq_ advances the commit_seq here and then the reading transactions assume it is committed. I will add a safety check after the commit to throw an exception if this case happened.